tsdb/head_append.go

```diff
@@ -458,6 +462,33 @@ func (a *headAppender) AppendHistogram(ref storage.SeriesRef, lset labels.Labels
 	return storage.SeriesRef(s.ref), nil
 }
 
+func validateHistogram(h *histogram.Histogram) error {
+	checkSpans := func(sign string, spans []histogram.Span, buckets []int64) error {
```

Comment:
We don't need to in-line this function here. And you can export ValidateHistogram method so the downstream users can use it as well.